bindings/matrix-sdk-ffi/src/notification_settings.rs

```diff
@@ -0,0 +1,277 @@
+        Self { client, push_rules }
+    }
+
+    async fn save_push_rules(&self, ruleset: &Ruleset) -> Result<(), ClientError> {
+        let content = PushRulesEventContent::new(ruleset.clone());
+        self.client.account().set_account_data::<PushRulesEventContent>(content).await?;
+        Ok(())
+    }
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl NotificationSettingsManager {
+
+    /// Get the notification mode for a given room
+    pub async fn get_room_notification_mode(&self, room_id: String) -> Result<RoomNotificationSettings, ClientError> {
+        // Get the actual mode defined by the user
+        if let Some(mode) = self.get_user_defined_room_notification_mode(&room_id) {
+            return Ok(RoomNotificationSettings::new(mode, false));
+        }
+
+        // If the user didn't defined a mode, return the default one for this room
+        let room_id = RoomId::parse(room_id)?;
+        let room = self.client.get_room(&room_id).context("Room not found")?;
+
+        let is_encrypted = room.is_encrypted().await?;
+        let is_direct = room.is_direct().await?;
+        let members_count = room.joined_members_count();
+        let default_mode = self.get_room_default_mode(is_encrypted, is_direct, members_count);
+        Ok(RoomNotificationSettings::new(default_mode, true))
+    }
+
+    /// Set a notification mode for a given room
+    pub async fn set_room_notification_mode(&self, room_id: String, mode: RoomNotificationMode) -> Result<(), ClientError> {
+        let mut new_ruleset = self.push_rules.clone();
+        match mode {
+            RoomNotificationMode::AllMessages => {
+                self.insert_notify_room_rule(&room_id, &mut new_ruleset)?;
+            },
+            RoomNotificationMode::MentionsAndKeywordsOnly => {
+                self.insert_mention_and_keywords_room_rule(&room_id, &mut new_ruleset)?;
+            },
+            RoomNotificationMode::Mute => {
+                self.insert_mute_room_rule(&room_id, &mut new_ruleset)?;
+            }
+        }
+        self.save_push_rules(&new_ruleset).await
+    }
+
+    /// Restore the default notification mode for a given room
+    pub async fn restore_default_room_notification_mode(&self, room_id: String) -> Result<(), ClientError> {
+        // Only if we have a custom mode for this room
+        match self.get_user_defined_room_notification_mode(&room_id) {
+            Some(..) => {
+                // get a new ruleset without the custom rules
+                let mut new_ruleset = self.push_rules.clone();
+                self.delete_user_defined_room_notification_mode(&room_id, &mut new_ruleset)?;
+                self.save_push_rules(&new_ruleset).await
+            },
+            None => Ok(())
+        }
+    }
+}
+
+impl NotificationSettingsManager {
+
+    pub fn get_user_defined_room_notification_mode(&self, room_id: &String) -> Option<RoomNotificationMode> {
```

Comment:
does this one need to be` pub`?

Reply:
Yes, you're right. I've removed the `pub`